workspaces/adventure-pack/goodies/python3/src/list_transpose/__init__.py

```diff
@@ -0,0 +1,16 @@
+from typing import Any
+
+
+def list_transpose(list: list[Any]) -> list[Any]:
+    r = len(list)
+    c = len(list[0]) if r != 0 else 0
+    if r == 0 or c == 0:
+        return []
+    if any(len(row) != c for row in list):
+        return []
+
+    ret = [[0] * r for _ in range(c)]
+    for i in range(c):
+        for j in range(r):
+            ret[i][j] = list[j][i]
+    return ret
```

Comment:
There's actually a really cheeky implementation of transpose that might be fun to use here. Take a look at https://stackoverflow.com/a/19888067

Reply:
Very clever, I like it.